modules/SCsub

```diff
@@ -1,17 +1,26 @@
 #!/usr/bin/env python
 
-Import("env")
-
 import modules_builders
 import os
 
+Import("env")
+
 env_modules = env.Clone()
 
 Export("env_modules")
 
 # Header with MODULE_*_ENABLED defines.
 env.CommandNoCache("modules_enabled.gen.h", Value(env.module_list), modules_builders.generate_modules_enabled)
 
+# Header to be included in `tests/test_main.cpp` to run module-specific tests.
+if env["tests"]:
+    env.CommandNoCache(
+        "modules_tests.gen.h",
+        Value(env.module_list),
+        Action(modules_builders.generate_modules_tests, "Generating modules tests header."),
+    )
+    env.AlwaysBuild("modules_tests.gen.h")
+
```

Comment:
Using `env.AlwaysBuild` because headers are not build objects, so changes to them are not detected by SCons, leading to wrong or out-of-date `modules_tests.gen.h`.

By default, `env.Command` or `env.CommandNoCache` print their output for each source. But wrapping the actions with `Action` objects makes it print only the name of the running command, preventing unnecessary verbosity each time Godot is build caused by `env.AlwaysBuild`.

In theory, I think it's possible to implement some custom `env.Decider(function)`, but I'm not sure whether it's worth doing so without adding too much complexity.